### DISCUSSION TITLE
Laravel nova eager loading model not working

### DISCUSSION THREAD

**Discussion by zsoltjanes** [spatie/laravel-medialibrary#3640] [Q&A]
Hi Guys! I use Laravel Nova, and as their documentum says i should load the relation with the `$with` array https://nova.laravel.com/docs/resources/#eager-loading
but its not working for me. 

The Nova/Media.php file $model is calling your basic modell class.

```
    public static $with = [
        'model',
    ];
    
    public static $model = \Spatie\MediaLibrary\MediaCollections\Models\Media::class;
```

Attempted to lazy load [announcement] on model [App\Models\Application] but lazy loading is disabled.

The Nova/Application model has the $with too:

```
    public static $with = [
        'user',
        'announcement',
    ];
```

**Comment by blaqdog** [comment]
Add `->with(<your relations>);` in your model `Nova\Resource` in `indexQuery()` method like this:
```
public static function indexQuery(NovaRequest $request, $query)
{
    return $query->with([
        'user',
        'announcement',
    ]);
}
```
